### ISSUE TITLE
wallet: bitcoind fails to auto-load a non-empty external-signer wallet

### ISSUE THREAD

**Issue by hebasto** [bitcoin/bitcoin#21919]
When a non-empty wallet with the `WALLET_FLAG_EXTERNAL_SIGNER` flag is supposed to be auto-loaded, i.e., it is mentioned in the `settings.json` file, and binaries are built with the `--disable-external-signer` configure option, the `bitcoind` fails:
```
...
2021-05-11T12:07:20Z [init] Using SQLite Version 3.31.1
2021-05-11T12:07:20Z [init] Using wallet /home/hebasto/.bitcoin/testnet3/wallets/coldcard_t
2021-05-11T12:07:20Z [init] init message: Loading wallet…
2021-05-11T12:07:20Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:07:20Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:07:20Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:07:20Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:07:20Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:07:20Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:07:20Z [init] [coldcard_t] Setting spkMan to active: id = c2fe2eddfccf102a74da940e282062adc8c13fbd28d00477644d3fcd1f6efe5d, type = 0, internal = 0
2021-05-11T12:07:20Z [init] [coldcard_t] Releasing wallet
2021-05-11T12:07:20Z [init] 

************************
EXCEPTION: St12out_of_range       
map::at       
bitcoin in AppInit()       



************************
EXCEPTION: St12out_of_range       
map::at       
bitcoin in AppInit()       

2021-05-11T12:07:20Z [init] Shutdown: In progress...
2021-05-11T12:07:20Z [scheduler] scheduler thread exit
2021-05-11T12:07:21Z [shutoff] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) started
2021-05-11T12:07:21Z [shutoff] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) completed (0.00s)
2021-05-11T12:07:21Z [shutoff] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) started
2021-05-11T12:07:21Z [shutoff] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) completed (0.00s)
2021-05-11T12:07:21Z [shutoff] [w20191120] Releasing wallet
2021-05-11T12:07:21Z [shutoff] [watch-only] Releasing wallet
2021-05-11T12:07:21Z [shutoff] [default wallet] Releasing wallet
2021-05-11T12:07:21Z [shutoff] Shutdown: done
```

The reason is the `CWallet::Create` throws an exception that is not caught in https://github.com/bitcoin/bitcoin/blob/e175a20769b5a7b98ee3082d89f9d4f31a4503d6/src/wallet/load.cpp#L117-L120

With the following diff
```diff
--- a/src/wallet/load.cpp
+++ b/src/wallet/load.cpp
@@ -117,6 +117,9 @@ bool LoadWallets(interfaces::Chain& chain)
     } catch (const std::runtime_error& e) {
         chain.initError(Untranslated(e.what()));
         return false;
+    } catch (const std::exception& e) {
+        chain.initError(Untranslated(e.what()));
+        return false;
     }
 }
 
```

the `bitcoind` output is:
```
...
2021-05-11T12:18:21Z [init] Using SQLite Version 3.31.1
2021-05-11T12:18:21Z [init] Using wallet /home/hebasto/.bitcoin/testnet3/wallets/coldcard_t
2021-05-11T12:18:21Z [init] init message: Loading wallet…
2021-05-11T12:18:21Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:18:21Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:18:21Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:18:21Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:18:21Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:18:21Z [init] [coldcard_t] LoadDescriptorScriptPubKeyMan: Compiled without external signing support (required for external signing)
2021-05-11T12:18:21Z [init] [coldcard_t] Setting spkMan to active: id = c2fe2eddfccf102a74da940e282062adc8c13fbd28d00477644d3fcd1f6efe5d, type = 0, internal = 0
2021-05-11T12:18:21Z [init] [coldcard_t] Releasing wallet
2021-05-11T12:18:21Z [init] Error: map::at
Error: map::at
2021-05-11T12:18:21Z [init] Shutdown: In progress...
2021-05-11T12:18:22Z [scheduler] scheduler thread exit
2021-05-11T12:18:22Z [shutoff] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) started
2021-05-11T12:18:22Z [shutoff] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) completed (0.00s)
2021-05-11T12:18:22Z [shutoff] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) started
2021-05-11T12:18:22Z [shutoff] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) completed (0.00s)
2021-05-11T12:18:23Z [shutoff] [w20191120] Releasing wallet
2021-05-11T12:18:23Z [shutoff] [watch-only] Releasing wallet
2021-05-11T12:18:23Z [shutoff] [default wallet] Releasing wallet
2021-05-11T12:18:23Z [shutoff] Shutdown: done
```

---

Noted while reviewing https://github.com/bitcoin-core/gui/pull/4.

**Comment by Rspigler** [comment]
Doesn't it need to be built with `--enable-external-signer` in order to load the wallet?

**Comment by hebasto** [comment]
> Doesn't it need to be built with `--enable-external-signer` in order to load the wallet?

If so, it is a UX flaw because the `--enable-external-signer` should not be required if a user is going to check his wallet without signing new transactions.

**Comment by Rspigler** [comment]
Hm, I don't know
@Sjors

**Comment by Sjors** [comment]
Presumably this applies to _any_ mandatory wallet flag that's not supported? See also https://github.com/bitcoin-core/gui/pull/236

**Comment by Sjors** [comment]
I added a commit that makes this behave better in #21935.

**Comment by meshcollider** [comment]
Fixed in #22173
